### PR TITLE
custom target: Consider all build depends while serializing

### PR DESCRIPTION
**Pull request by nirbheek**
Currently, we only consider the build depends of the `Executable` being run when serializing custom targets. However, this is not always sufficient, for example if the executable loads modules at runtime or if the executable is actually a python script that loads a built module.

For these cases, we need to set `PATH` on Windows correctly or the custom target will fail to run at build time complaining about missing DLLs.

### CODE REVIEW

File: mesonbuild/backend/vs2010backend.py [hunk 1]
```diff
@@ -429,9 +429,12 @@ def gen_custom_target_vcxproj(self, target, ofname, guid):
         # Always use a wrapper because MSBuild eats random characters when
         # there are many arguments.
         tdir_abs = os.path.join(self.environment.get_build_dir(), self.get_target_dir(target))
+        extra_bdeps = target.get_transitive_build_target_deps()
+        extra_paths = self.determine_windows_extra_paths(target.command[0], extra_bdeps):
```
Comment (ignatenkobrain) [inline mesonbuild/backend/vs2010backend.py]: __[Flake8]__

SyntaxError: invalid syntax

Comment (ignatenkobrain) [inline mesonbuild/backend/vs2010backend.py]: __[Flake8]__

SyntaxError: invalid syntax

**Comment by nirbheek** [comment]
Test failure seems to be unrelated.

**Comment by nirbheek** [comment]
Actually, the msys2-mingw x86 failure may not be spurious. Probably linking against the wrong Python libraries?

**Comment by nirbheek** [comment]
So strange, I cannot reproduce this when I run the test manually in my own 64-bit MinGW environment with 64-bit Python, which is what the failure mode is on the CI.

**Comment by nirbheek** [comment]
This is ready to go in now. I fixed the test failure by making the MSYS2-MinGW target use the MSYS2 Python 3.

**Comment by jibsen** [comment]
Did you try running the tests through the bash provided with MSYS2?

**Comment by nirbheek** [comment]
I did all my testing inside MSYS2's MinGW64 and 32 shells, and I also tested inside cmd.exe

**Pull request merged**